src/sota_tools/deploy.cc

```diff
@@ -168,24 +168,34 @@ bool copy_repo(const std::string &cacerts, const std::string &src, const std::st
   if (errors) {
     LOG_ERROR << "One or more errors while pushing";
     return EXIT_FAILURE;
-  } else {
-    return EXIT_SUCCESS;
   }
 
+  if (sign) {
+    sign_repo(dst, ostree_ref, hardwareids);
+  }
```

Comment:
Is there a reason why `copy_repo` calls `sign_repo`?

Reply:
It is because `ostree_ref` argument is creating inside of `copy_repo` and it contains branch name and hash calculation function that is needed inside sign_repo.  Such design doesn't allow to do that better.

Reply:
Ah right, I see now that `ref` is of a different type in either cases.